x-pack/plugins/apm/server/lib/service_map/transform_service_map_responses.ts

```diff
@@ -113,7 +137,8 @@ export function transformServiceMapResponses(response: ServiceMapResponse) {
           },
           ...matchedServiceNodes.map((serviceNode) =>
             pickBy(serviceNode, identity)
-          )
+          ),
+          serviceAnomalyStats ? { serviceAnomalyStats } : null
```

Comment:
This whole `Object.assign` thing is pretty hard to grok at first. Can we simplify it to something like this:

```ts
    if (matchedServiceNodes.length) {
      return {
        ...map,
        [node.id]: {
          id: matchedServiceNodes[0][SERVICE_NAME],
          serviceAnomalyStats,
          ...matchedServiceNodes,
        },
      };
    }
```
_Note: above is not an exact replacement_